src/org/openstreetmap/josm/gui/tagging/presets/TaggingPresetSelector.java

```diff
@@ -64,11 +65,15 @@ public class TaggingPresetSelector extends SearchTextResultListPanel<TaggingPres
 
     private static final Pattern PATTERN_PUNCTUATION = Pattern.compile("\\p{Punct}", Pattern.UNICODE_CHARACTER_CLASS);
 
-    private static final BooleanProperty SEARCH_IN_TAGS = new BooleanProperty("taggingpreset.dialog.search-in-tags", true);
-    private static final BooleanProperty ONLY_APPLICABLE = new BooleanProperty("taggingpreset.dialog.only-applicable-to-selection", true);
+    Map<PresetSearchFilter, Boolean> ckpreferences = new EnumMap<>(PresetSearchFilter.class);
+    {
+        //initial value of the checkboxes
+        ckpreferences.put(PresetSearchFilter.ONLY_APPLICABLE, true);
+        ckpreferences.put(PresetSearchFilter.SEARCH_IN_TAGS, true);
+        ckpreferences.put(PresetSearchFilter.DEPRECATED_TAGS, false);
+    }
```

Comment:
We should probably keep the properties.

Realistically, the map can be populated with `ckpreferences.computeIfAbsent(key, k -> PROPERTY.get())`.

Using (and saving) the preferences lets us keep the checkbox state between application restarts.